### ISSUE TITLE
docker image ?

### ISSUE THREAD

**Issue by UniqueFool** [arrayfire/arrayfire#1439]
According to github, there have been several attempts at providing a docker image for arrayfire, e.g.: https://github.com/ozzyjohnson/docker-arrayfire

Many of these are meanwhile outdated - and it used to be really tedious to provide such images because nvidia didn't provide OpenCL support - as of 2 weeks ago, there is now experimental OpenCL support for nvidia docker images: https://github.com/NVIDIA/nvidia-docker/commit/9fdb5f63c66fd31b08d2976230699b4661cdf0d9

In other words, it would now be possible to deploy arrayfire, including most/all of its optional bindings, as a single docker image.

**Comment by vakopian** [comment]
I just created a Dockerfile that compiles the current version of ArrayFire (3.4.1) on ubuntu 14.04. See below
```
FROM ubuntu:14.04.5

RUN apt-get update
RUN apt-get -y install wget software-properties-common
RUN wget http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-repo-ubuntu1604_8.0.44-1_amd64.deb
RUN dpkg -i cuda-repo-*amd64.deb
# add the repository for libglfw3
RUN apt-add-repository --yes ppa:keithw/glfw3
# add the repository for cmake 3 (required by parts of arrayfire)
RUN apt-add-repository --yes ppa:george-edison55/cmake-3.x
RUN apt-get update

# install some common packages
RUN apt-get install -y wget curl openssh-client bash-completion vim emacs24-nox zip unzip nano command-not-found gdb
# install packages needed to build ArrayFire
RUN apt-get install -y build-essential git cmake libfreeimage-dev cmake-curses-gui libatlas3gf-base libatlas-dev libfftw3-dev liblapacke-dev libboost-all-dev opencl-headers libglfw3-dev libfontconfig1-dev

# install cuda, exlude some packages that cause problems for apt-get under docker
RUN apt-get install -y cuda unity-control-center- unity-settings-daemon- zenity- gnome-user-guide- yelp-

# some aliases and bash goodies
RUN echo "alias ll='ls -alF'" >> /root/.bashrc && echo "if [ -f /etc/bash_completion ]; then source /etc/bash_completion; fi" >> /root/.bashrc
RUN /bin/echo -e '"\\eOA": history-search-backward\n"\\eOB": history-search-forward\n"\\eOC": forward-char\n"\\eOD": backward-char\n"\\e[A": history-search-backward\n"\\e[B": history-search-forward\n"\\e[C": forward-char\n"\\e[D": backward-char' > /root/.inputrc


RUN git clone https://github.com/arrayfire/arrayfire.git && cd arrayfire && git submodule init && git submodule update && git checkout v3.4.1
RUN cd /arrayfire && mkdir build && cd build && cmake .. -DCMAKE_BUILD_TYPE=Release && make -j8
```

Perhaps it can be cleaned-up by the owners and made part of the repo so it can be maintained with each new version.

**Comment by shehzan10** [comment]
I think the plan is to have one docker that can compile ArrayFire from source and another that can use the ArrayFire installer. So you have done the former.
I have assigned this issue to @mlloreda who will look into it from our end.

**Comment by mlloreda** [comment]
Okay guys,

Official `Dockerfile` is over at [arrayfire-docker](https://github.com/arrayfire/arrayfire-docker). We've also pushed an image to [Docker Hub](https://hub.docker.com/r/arrayfire/arrayfire/) - hoping to add tags soon.

The `Dockerfile` imports from Nvidia's CUDA 8 image, so be sure to use the `nvidia-docker` binary instead of vanilla `docker`.

**Comment by 9prady9** [comment]
Closing the issue since the feature request has been addressed.
